test/e2e/test/beat/builder.go

```diff
@@ -31,9 +31,9 @@ const (
 
 // Builder to create a Beat
 type Builder struct {
-	Beat        beatv1beta1.Beat
-	Validations []ValidationFunc
-	RBACObjects []runtime.Object
+	Beat              beatv1beta1.Beat
+	Validations       []ValidationFunc
+	AdditionalObjects []runtime.Object
```

Comment:
Maybe I'm missing something but I can't find where these `AdditionalObjects` are created/deleted.

Reply:
Through `append` in `WithObjects` and `bind` funcs, is that what you were asking?

Reply:
I'm wondering where the k8s client is called for these objects. I'm not able to run the e2e tests locally:

```
 {"service.version": "1.1.0-SNAPSHOT-mmo", "controller": "beat-controller", "request": "e2e-mercury/test-beat-configref-g6t2", "error": "secret fb-config not found in e2e-mercury namespace", "errorCauses": [{"error": "secret fb-config not found in e2e-mercury namespace"}]}
```

Reply:
We should return them in `RuntimeObjects` func and they should be created/deleted in the cluster with `CreationTestSteps` and `DeletionTestSteps`. I'll try running locally to see if I can repro.